test/e2e/operator/deployment_api.go

```diff
@@ -0,0 +1,612 @@
+/*
+Copyright 2019 Intel Corporation.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package operator
+
+import (
+	"fmt"
+
+	"github.com/intel/pmem-csi/pkg/apis"
+	api "github.com/intel/pmem-csi/pkg/apis/pmemcsi/v1alpha1"
+	pmemtls "github.com/intel/pmem-csi/pkg/pmem-csi-operator/pmem-tls"
+
+	corev1 "k8s.io/api/core/v1"
+	storagev1beta1 "k8s.io/api/storage/v1beta1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	pmemDeploymentRes = schema.GroupVersionResource{
+		Group:    api.SchemeGroupVersion.Group,
+		Version:  api.SchemeGroupVersion.Version,
+		Resource: "deployments",
+	}
+
+	scheme = runtime.NewScheme()
+)
+
+var _ = DescribeForAll("pmem-csi", func(o *Operator, f *framework.Framework) {
```

Comment:
The name "pmem-csi" is redundant and has no meaning because the entire test suite is for PMEM-CSI.

Perhaps "operator"? But that really depends on what the full test string then becomes, which depends a bit on the integration into the test suite.

Reply:
I thought that the naming would be meaningful if combined with the inner context: "pmem-csi deployment".

Reply:
But "pmem-csi" doesn't tell us what kind of deployment we deal with. We have lots of different deployments now (driver in various modes, operator).

For example, one full test name (can be seen in https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/view/pmem-csi/job/pmem-csi/view/change-requests/job/PR-559/25/execution/node/92/log/?consoleFull) is now:
`pmem-csi deployment with defaults`.

That doesn't tell us that the test is about the operator.

But don't bother changing it. I'll probably address this when integrating it into the `deploy` package.